src/panels/Menu/TopMenuBar/TopMenuBar.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Button, Group, Kbd, Menu, Text } from '@mantine/core';
+import { modals } from '@mantine/modals';
+
+import { useGetBoolPropertyValue, useOpenSpaceApi } from '@/api/hooks';
+import {
+  AddFileIcon,
+  ConsoleIcon,
+  ExitAppIcon,
+  SaveIcon,
+  VisibilityIcon
+} from '@/icons/icons';
+import { WindowLayoutOptions } from '@/windowmanagement/WindowLayout/WindowLayout';
+
+import { HelpMenu } from './HelpMenu';
+import { NewWindowMenu } from './NewWindowMenu';
+import { TaskBarMenuChoices } from './TaskBarMenuChoices';
+import { MenuWrapper } from './MenuWrapper';
+
+interface TopMenuBarProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+  addWindow: (component: React.JSX.Element, options: WindowLayoutOptions) => void;
+}
+
+export function TopMenuBar({
+  visibleMenuItems,
+  setVisibleMenuItems,
+  addWindow
```

Comment:
This seems to only be passed as a prop - use the provider instead where the addWindow function is needed?